src/utils/components/SelectMultiTypeahead/Toggle.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {
+  Dispatch,
+  FC,
+  FormEvent,
+  KeyboardEvent,
+  Ref,
+  RefObject,
+  SetStateAction,
+} from 'react';
+
+import {
+  Button,
+  ButtonVariant,
+  Chip,
+  ChipGroup,
+  MenuToggle,
+  MenuToggleElement,
+  SelectOptionProps,
+  TextInputGroup,
+  TextInputGroupMain,
+  TextInputGroupUtilities,
+} from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { isEmpty } from '@utils/utils';
+
+import { NO_RESULTS_VALUE } from './constants';
+import { handleMenuArrowKeys } from './utils';
+
+type ToggleProps = {
+  focusedItemIndex: null | number;
+  inputValue: string;
+  isOpen: boolean;
+  onSelect: (value: string) => void;
+  placeholder: string;
+  selected: string[];
+  selectOptions: SelectOptionProps[];
+  setFocusedItemIndex: (newValue: null | number) => void;
+  setInputValue: (newInput: string) => void;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  setSelected: (newSelection: string[]) => void;
+  textInputRef: RefObject<any>;
+  toggleRef: Ref<MenuToggleElement>;
+};
+
+const Toggle: FC<ToggleProps> = ({
+  focusedItemIndex,
+  inputValue,
+  isOpen,
+  onSelect,
+  placeholder,
+  selected,
+  selectOptions,
+  setFocusedItemIndex,
+  setInputValue,
+  setIsOpen,
+  setSelected,
+  textInputRef,
+  toggleRef,
+}) => {
+  const { t } = useNetworkingTranslation();
+
+  const onInputKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    const enabledMenuItems = selectOptions.filter((menuItem) => !menuItem.isDisabled);
+    const [firstMenuItem] = enabledMenuItems;
+    const focusedItem = focusedItemIndex ? enabledMenuItems[focusedItemIndex] : firstMenuItem;
+
+    switch (event.key) {
+      // Select the first available option
+      case 'Enter':
+        if (!isOpen) {
+          setIsOpen((prevIsOpen) => !prevIsOpen);
+          return;
+        }
+
+        if (isOpen && focusedItem.value !== NO_RESULTS_VALUE) {
+          onSelect(focusedItem.value as string);
+          return;
+        }
+
+        break;
+      case 'Tab':
+      case 'Escape':
+        setIsOpen(false);
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+        event.preventDefault();
+
+        if (isOpen) {
+          const indexToFocus = handleMenuArrowKeys(event.key, focusedItemIndex, selectOptions);
+
+          if (indexToFocus !== null) setFocusedItemIndex(indexToFocus);
+        }
+        break;
+    }
+  };
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onTextInputChange = (_event: FormEvent<HTMLInputElement>, value: string) => {
+    setInputValue(value);
+    setFocusedItemIndex(null);
+
+    if (value) setIsOpen(true);
+  };
+
+  return (
+    <MenuToggle
+      innerRef={toggleRef}
+      isExpanded={isOpen}
+      isFullWidth
+      onClick={onToggleClick}
+      variant="typeahead"
```

Comment:
do we have a const variant from patternfly from it?

Reply:
no we don't have it.
https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/components/MenuToggle/MenuToggle.tsx#L39